src/OpenCV_Mat.jl

```diff
@@ -3,70 +3,106 @@
 
 cxx"""
 // Grayscale and binary images 
-template <typename T1>
-inline T1 ptr_val(cv::Mat<T1> &img, int row, int col)
+template <typename T>
+inline T ptr_val(cv::Mat &img, int row, int col)
 {
-  return(static_cast<int>(img.at<T1>(row, col)));
+  return(static_cast<int>(img.at<T>(row, col)));
 }
 
-template <typename T2>
-inline void ptr_val(cv::Mat<T2> &img, int row, int col, double val)
+template <typename T>
+inline void ptr_val(cv::Mat &img, int row, int col, double val)
```

Comment:
I have had a look at the possible implementation of `cv::Mat` templates as well, but as discussed before the implementation here does not work.  When executed in the REPL, I get `couldn't infer template argument T`:
 
```julia
julia> @cxx ptr_val(img, 10,10, 200)
:1:1: error: no matching function for call to 'ptr_val'
__cxxjl_93.cpp:2:13: note: candidate template ignored: couldn't infer template argument 'T'
inline void ptr_val(cv::Mat &img, int row, int col, double val) {
```
I will add some comments on my observations and suggestion for an implementation using a combination of `std::vector` and `_InputArray`.  `OpenCV`by design (due to performance reasons) uses non-templated `Mat` constructors.